Development/src/pages/devices/ChannelMappingMatrix.js

```diff
@@ -48,129 +62,321 @@ const TooltipDivider = withStyles({
     },
 })(Divider);
 
-const getOutputTooltip = (outputId, outputItem, io) => {
+const InteractiveTooltip = ({
+    getTooltip,
+    display,
+    tooltipOpen,
+    setTooltipOpen,
+}) => {
+    const [displayEditTextField, setDisplayEditTextField] = useState(false);
+    const [open, setOpen] = useState(false);
+    const handleClose = () => {
+        if (open && !displayEditTextField) {
+            setOpen(false);
+            setTooltipOpen(false);
+        }
+    };
+
+    const handleOpen = () => {
+        if (!tooltipOpen) {
+            setOpen(true);
+            setTooltipOpen(true);
+        }
+    };
+
     return (
-        <>
-            {'ID'}
-            <Typography variant="body2">{outputId}</Typography>
-            {'Name'}
-            <Typography variant="body2">
-                {outputItem.properties.name}
-            </Typography>
-            {'Description'}
-            <Typography variant="body2">
-                {outputItem.properties.description}
-            </Typography>
-            <TooltipDivider />
-            {'Routable Inputs'}
-            <Typography variant="body2">
-                {outputItem.caps.routable_inputs !== null
-                    ? outputItem.caps.routable_inputs
-                          .map(inputId => {
-                              return inputId === null
-                                  ? 'Unrouted'
-                                  : get(
-                                        io,
-                                        `inputs.${inputId}.properties.name`
-                                    );
-                          })
-                          .join(', ')
-                    : 'No Constraints'}
-            </Typography>
-        </>
+        <Tooltip
+            open={open}
+            interactive
+            title={getTooltip(displayEditTextField, setDisplayEditTextField)}
+            placement="bottom"
+            onOpen={handleOpen}
+            onClose={handleClose}
+        >
+            <div>{display()}</div>
+        </Tooltip>
     );
 };
 
-const getInputTooltip = (inputId, inputItem) => {
-    return (
-        <>
-            {'ID'}
-            <Typography variant="body2">{inputId}</Typography>
-            {'Name'}
-            <Typography variant="body2">{inputItem.properties.name}</Typography>
-            {'Description'}
-            <Typography variant="body2">
-                {get(inputItem, `properties.description`)}
-            </Typography>
-            <TooltipDivider />
-            {'Block Size'}
-            <Typography variant="body2">
-                {get(inputItem, `caps.block_size`)}
-            </Typography>
-            {'Reordering'}
-            <Typography variant="body2">
-                {get(inputItem, `caps.reordering`) ? (
-                    <DoneIcon />
-                ) : (
-                    <ClearIcon />
-                )}
-            </Typography>
-        </>
-    );
+const getOutputTooltip = (
```

Comment:
Similar to comment about `display` function in `InteractiveTooltip`, I think idiomatically this and the other `get...Tooltip` functions  would be a React component called e.g. `OutputTooltipContent`.